### PR TITLE
Bluetooth: controller: Added support for vendor ticker nodes

### LINKED ISSUE #25067: Insufficient ticker nodes for vendor implementations
Vendors need to be able to reserve ticker nodes for proprietary LL implementations. There is currently no way of achieving this. Dynamically reusing ticker ids reserved by ULL may cause erratic behavior.

**Impact**
Reusing other ticker ids dyamically is not a safe way to implement features, as usage or rules may change in upstream code. A number of user- or vendor ticker nodes should be configurable.

### PR DESCRIPTION
**Pull request by mtpr-ot**
With BT_CTLR_USER_TICKER_ID_RANGE it is possible for vendors to add a
number of ticker nodes for proprietary purposes. The feature depends on
BT_CTLR_USER_EXT.

Signed-off-by: Morten Priess <mtpr@oticon.com>

Fixes #25067

### CODE REVIEW

File: subsys/bluetooth/controller/Kconfig [hunk 1]
```diff
@@ -882,6 +882,15 @@ config BT_CTLR_USER_EVT_RANGE
 	  Number of event types reserved for proprietary use. The range
 	  is typically used when BT_CTLR_USER_EXT is in use.
 
+config BT_CTLR_USER_TICKER_ID_RANGE
+	int "Range of ticker id constants reserved for proprietary ticker nodes"
+	depends on BT_CTLR_USER_EXT
+	default 0
+	range 0 10
```
Comment (thoh-ot) [inline subsys/bluetooth/controller/Kconfig:889]: Do we need to add a range limitation on this?

File: subsys/bluetooth/controller/ll_sw/lll.h [hunk 2]
```diff
@@ -54,6 +54,12 @@ enum {
 			       1),
 #endif /* CONFIG_BT_CONN */
 
+#if defined(CONFIG_BT_CTLR_USER_EXT)
+	TICKER_ID_USER_BASE,
+	TICKER_ID_USER_LAST = (TICKER_ID_USER_BASE +
+			       CONFIG_BT_CTLR_USER_TICKER_ID_RANGE - 1),
+#endif /* CONFIG_BT_CTLR_USER_EXT */
+
```
Comment (thoh-ot) [inline subsys/bluetooth/controller/ll_sw/lll.h:62]: Doesn't this block only make sense if CONFIG_BT_CTLR_USER_TICKER_ID_RANGE > 0?

**Review by thoh-ot — CHANGES_REQUESTED** [review]

**Review by cvinayak — APPROVED** [review]

File: subsys/bluetooth/controller/Kconfig (hunk 1 above)
Comment (mtpr-ot) [reply subsys/bluetooth/controller/Kconfig:889]: > Do we need to add a range limitation on this?

No we don't need the range.

File: subsys/bluetooth/controller/ll_sw/lll.h (hunk 2 above)
Comment (mtpr-ot) [reply subsys/bluetooth/controller/ll_sw/lll.h:62]: > Doesn't this block only make sense if CONFIG_BT_CTLR_USER_TICKER_ID_RANGE > 0?

Correct. Should we add the condition CONFIG_BT_CTLR_USER_TICKER_ID_RANGE > 0 to line 57?

Comment (thoh-ot) [reply subsys/bluetooth/controller/ll_sw/lll.h:62]: > > Doesn't this block only make sense if CONFIG_BT_CTLR_USER_TICKER_ID_RANGE > 0?
> 
> Correct. Should we add the condition CONFIG_BT_CTLR_USER_TICKER_ID_RANGE > 0 to line 57?

That would make sense.

**Review by cvinayak — APPROVED** [review]

**Review by thoh-ot — APPROVED** [review]

**Comment by mtpr-ot** [comment]
Shippable failures are not from my commit...

**Review by carlescufi — APPROVED** [review]

**Pull request merged**